core/csv_api.php

```diff
@@ -438,3 +438,35 @@ function csv_format_due_date( BugData $p_bug ) {
 function csv_format_sponsorship_total( BugData $p_bug ) {
 	return csv_escape_string( $p_bug->sponsorship_total );
 }
+
+/**
+ * return the attachment count for an issue
+ * @param BugData $p_bug A BugData object.
+ * @return string
+ * @access public
+ */
+function csv_format_attachment_count( BugData $p_bug ) {
+	# Check for attachments
+	$t_attachment_count = 0;
+	if( file_can_view_bug_attachments( $p_bug->id, null ) ) {
```

Comment:
My main point is to not having the calling scripts/apis worry about access checks.  Whatever access checks that we already support should live in the APIs of the resource being called.  For examples, bugnote_api apis should apply all access checks rather than csv_api/excel_api.  I was just giving examples of access checks, and not advocating implementing them as part of this PR.